### PR TITLE
Do not set pin_auth_protocol, if no pin_auth is available

### PR DESCRIPTION
**Pull request by msirringhaus**
My Yubikey did not like the pin_auth_protocol, if no PIN is set on it.

### CODE REVIEW

**Comment by jschanck** [comment]
In the case where `info.options.client_pin == Some(false)` our behavior here should depend on whether user verification is preferred / required by the RP. I think you can safely fold `calculate_pin_auth` into the default implementation of `PinAuthCommand::determine_pin_auth` to simplify the logic. If the RP requires uv we should return `AuthenticationError::PinNotSet` and prompt the user.

**Comment by msirringhaus** [comment]
Folding the two functions results in a bit more error-handling clutter, which is why I kept them separate (because of the Pin-error remapping, which I would need to do in multiple places). But I moved the pin_auth_protocol-logic into `calculate_pin_auth`. Let me know if you want to fold the functions, anyways.

Regarding `PinNotSet`: The token does this check for us. It will return with `PinNotSet`. 
Until now I didn't want to reproduce all the checks the token does in our code as well, as IMHO that is rather error-prone. I have the check for PinRequired there, to make the code easier. Otherwise I let the token decide what it accepts and what it does not.
But adding a StatusUpdate for it makes sense and was a TODO anyways. Then on FF side we can display the appropriate error message.

File: src/ctap2/commands/mod.rs [hunk 1]
```diff
@@ -103,13 +103,14 @@ pub(crate) trait PinAuthCommand {
...
         let client_data_hash = self.client_data_hash();
-        let pin_auth = match calculate_pin_auth(dev, &client_data_hash, &self.pin()) {
-            Ok(pin_auth) => pin_auth,
-            Err(e) => {
-                return Err(repackage_pin_errors(dev, e));
-            }
-        };
-        self.set_pin_auth(pin_auth, Some(1)); // TODO(MS): Currently, we only support version 1
+        let (pin_auth, pin_auth_protocol) =
+            match calculate_pin_auth(dev, &client_data_hash, &self.pin()) {
```
Comment (jschanck) [inline src/ctap2/commands/mod.rs:107]: nit: I think `.map_err(|e| repackage_pin_errors(dev, e))?` is more idiomatic here.

**Review by jschanck — APPROVED** [review]
Looks good. And I confirmed that this resolves [Bug 1804624](https://bugzilla.mozilla.org/show_bug.cgi?id=1804624).

**Pull request merged**